uw-frame-components/portal/main/services.js

```diff
@@ -15,6 +15,16 @@ define(['angular'], function(angular) {
 
       userPromise = prom.then(
           function(data, status) { //success function
+            if(APP_FLAGS.loginOnLoad) {
+              //quick check to make sure you are who your browser says you are
+              if($sessionStorage.portal
+                && $sessionStorage.portal.username
+                && data.data.person.userName != $sessionStorage.portal.username) {
+                  console.warn("Thought they were " + $sessionStorage.portal.username +
+                   " but session sent back " + data.data.person.userName +". Redirect!");
+                  miscService.redirectUser(302, "Wrong User than populated in session storage.");
```

Comment:
Is this redirect sufficient cleanup, or should more be done here to clear sessionStorage?